### PR TITLE
Add cmake and corresponding toolchain files, add zip, fix $PATH for devkitA64 to work with cmake

### PR DESCRIPTION
**Pull request by rsn8887**
This allows building Switch homebrew with cmake, and adds zip. Zip is often used to package homebrew on the Switch.

Fixes https://github.com/devkitPro/docker/issues/3

### CODE REVIEW

**Comment by rsn8887** [comment]
Note: this makes building Switch stuff "out-of-the" box possible. Before this, $Path to gnu compiler drivers was not set, cmake was not there, zip was missing.

File: devkita64/Dockerfile [hunk 1]
```diff
@@ -4,5 +4,7 @@ MAINTAINER Dave Murphy <davem@devkitpro.org>
 
 RUN dkp-pacman -Syyu --noconfirm switch-dev && \
     dkp-pacman -S --needed --noconfirm `dkp-pacman -Slq dkp-libs | grep '^switch-'` && \
-    dkp-pacman -Scc --noconfirm
+    dkp-pacman -Scc --noconfirm && \
+    dkp-pacman -S --needed --noconfirm devkitpro-pkgbuild-helpers
 
+ENV PATH=$DEVKITPRO/devkitA64/bin:$PATH
```
Comment (WinterMute) [inline devkita64/Dockerfile]: devkitA64 should not be in global path. If we do this it then encourages sloppy projects that will cause problems for users. We've already seen this happen with people having multiple arm-none-eabi toolchains installed and attempting to use the wrong one for the wrong project.

Comment (rsn8887) [reply devkita64/Dockerfile]: How to make it work with switch.cmake then?

Comment (WinterMute) [reply devkita64/Dockerfile]: No. The point of switch.cmake is to help compiling ports which require cmake. The point of our Makefile based templates is to allow users to just run make & not care about any setup before building a project.

Currently we're sourcing switchvars.sh before running cmake, like this -> https://github.com/devkitPro/pacman-packages/blob/9128546582338c06d4fc11160cc04081823bc310/switch/bulletphysics/PKGBUILD#L20

There is some more discussion on updating the cmake support files here - https://github.com/devkitPro/pacman-packages/pull/64

Like I said on IRC, this is all a WiP and many things still need to be done despite the many things that have already been done.

Comment (rsn8887) [reply devkita64/Dockerfile]: Ok, understood.

**Pull request closed without merging**